### PR TITLE
Issue #9: Request Model

### PR DESCRIPTION
**Pull request by jtingling**
### What this PR does (required):
- A model for storing data about a dog sitting request.

### Any information needed to test this feature (required):
- Create a new document by providing: `userId`, `sitterId`, `date`, `start`, `end`, properties to a model create function.

### CODE REVIEW

File: server/models/Request.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+const mongoose = require("mongoose");
+
+const requestSchema = new mongoose.Schema({
+    userId: {
+        type: mongoose.Schema.Types.ObjectId,
+        required: true,
+    },
+    sitterId: {
+        type: mongoose.Schema.Types.ObjectId,
+        required: true,
```
Comment (ezeilo-su) [inline server/models/Request.js:10]: Add a `ref` to `User` model

**Review by sundayezeilo — CHANGES_REQUESTED** [review]
Looks good. Just a few comments to address, though. :smile:

File: server/models/Request.js (hunk 1 above)
Comment (jtingling) [reply server/models/Request.js:10]: I'll add the ref on next commit.

File: server/models/Request.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+const mongoose = require("mongoose");
+
+const requestSchema = new mongoose.Schema({
+    userId: {
+        type: mongoose.Schema.Types.ObjectId,
+        required: true,
```
[the remark this answers is not in the document]
Comment (jtingling) [reply server/models/Request.js:6]: I'll add the ref on next commit.

File: server/models/Request.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        required: true
+    },
+    start: { 
+        type: String, 
+        required: true,
+    },
+    end: { 
+        type: String, 
+        required: true,
+    },
```
Comment (moffatethan) [inline server/models/Request.js:25]: Let's remove these for now

**Review by moffatethan — CHANGES_REQUESTED** [review]
Just some changes on the code style - overall, good work. Not sure why the controller work is in this ticket. But we can merge the two together.

File: server/controllers/request.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    .exec(function(err, requests) {
+        if (err) {
+            res.status(500);
+            throw new Error('Problem finding document.')};
+        if (!requests) {
+            res.status(404);
+            throw new Error("No request for user exists.");
+        }
+        res.status(200).json({requests});
+    });
```
[the remark this answers is not in the document]
Comment (jtingling) [reply server/controllers/request.js:16]: > Just some changes on the code style - overall, good work. Not sure why the controller work is in this ticket. But we can merge the two together.

I realized I mistakenly committed the controller work into this PR so I tried to revert back one commit, stashed the changes for controller and made a new branch off develop and reapplied the stashed commit to the new branch for #45.  This is why both #44 and #45 include duplicate code.

**Review by moffatethan — CHANGES_REQUESTED** [review]
Please address the merge conflicts

**Comment by jtingling** [comment]
The merge conflict has been resolved.

**Review by moffatethan — APPROVED** [review]

**Pull request merged**